### PR TITLE
Add docs about pipeline upload, add Advanced Features subsection

### LINKED ISSUE #651: [FEATURE] Allow power users to add/modify pipelines
**Is your feature request related to a problem? If so, Please describe.**
I'd like to run `utilities` -> `add_segmentation_watershed` with a parameter changed from the default. It would be nice to do this on VIAME web because this is a short experiment that would be nice to visualize in browser. However, after consulting the documentation, it doesn't seem like I can run any pipelines other than those which are provided. 

**Describe the solution you'd like**
The solution that seems most appealing to me is a mechanism which allows users to upload pipelines to the user directory and have them become visible in the `Run Pipeline` menu. 

**Describe alternatives you've considered**
* I could run this pipeline in my local VIAME build and this is likely what I'll do short term. However, setting up VIAME might not be super easy, even for a technical user. Additionally, it's visualization capabilities are quite poor and this is my main complaint.
* I could try to configure DIVE locally which would likely get me good visualization. However, trying to set up Docker locally just for this seems like a bit of a hassle.

**Additional context**
I don't think this is super critical, but could make DIVE more useful to people with a moderate technical background.

### PR DESCRIPTION
**Pull request by subdavis**
Add docs about pipeline upload.

Hide CLI, Server Operator, and Pipeline Upload under an "Advanced Features" heading to dissuade casual readers from confusing these pages for real user guide pages.

fixes #651

### CODE REVIEW

**Comment by subdavis** [comment]
Turns out we also need to support requiring input detections for these user-uploaded pipelines.  I think some simple metadata would work for this in lieu of the full schema thing being ready (or even as a permanent fallback, Idk).

**Comment by subdavis** [comment]
Turns out multi-directory includes were supported all along: https://github.com/VIAME/VIAME/issues/131

We could have been pulling addons into some separate directory without having to copy over the `common_*` pipes.  I might see if I can clean this up.

**Comment by subdavis** [comment]
This PR now has some additional changes that allow user upload of any `.pipe` file.

* It is no longer necessary to rename the file `detector.pipe`
* The truthiness check for `trained_pipeline` is looser now.
* The pipe can safely `include` from common or addon pipelins
* The pipeline can specify that it takes input with `requires_input` truthy check.

I have specifically tested and you can upload just the `utility_add_segmentations_watershed.pipe` and tag it properly and it runs.

File: docs/Pipeline-Upload.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+!!! warning
+    This feature is not yet standardized, and the instructions below may change.
+
+1. Open the girder interface at `/girder` and create a new private folder called `MyPipelines`
+    1. For our demo instance, open [https://viame.kitware.com/girder](https://viame.kitware.com/girder)
+1. Create a new folder in that private folder, and give it a name you'd like to associate with your new pipeline.
+1. Upload one or more files inside your new pipeline subfolder:
+    1. A pipeline file ending in the `.pipe` file extension
+    1. Whatever other model `.zip` files are required by the pipe, named exactly as the appear in your `.pipe` file above.
```
Comment (BryonLewis) [inline docs/Pipeline-Upload.md:13]: I think you want it named as it `they appear inside your` 
typically this is `trained_detector.zip` or something like that for trained detectors.

**Review by BryonLewis — CHANGES_REQUESTED** [review]
Minor docs edits and some questions about mixing user and `utility` pipes that require input.

Ran it through most everything I could think that was touched.  Created tracks with the default utilities, and uploaded ones, did training, imported default pipelines and imported custom trained ones from desktop as well.

File: docs/Pipeline-Upload.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+![Upload Pipeline](images/Misc/UploadPipeline.png)
+
+## Accepting input
+
+If your pipe must accept input, set the **pipeline folder** metadata property `requires_input` to `true`.
+
+## Including base pipelines
+
+User-uploaded pipelines may `include` any pipe already installed from the base image or an addon. Including other user-uploaded pipes is not supported.
```
[the remark this answers is not in the document]
Comment (subdavis) [reply docs/Pipeline-Upload.md:25]: This is referring to the `include` keyword in pipe files.  You cannot upload or use a pipe file that depends on another pipe file that was also uploaded.

[the remark this answers is not in the document]
Comment (BryonLewis) [reply docs/Pipeline-Upload.md:25]: ahh I didn't realize `include` was inside the context of a .pipe.

[the remark this answers is not in the document]
Comment (russelldj) [reply docs/Pipeline-Upload.md:25]: I'm not sure if this is at all useful in this context but `kwiver pipe-config -o` allows you to export a pipeline with all the content from `include`d files in one file. Perhaps documenting this is a good stopgap before supporting includes on the user side.

[the remark this answers is not in the document]
Comment (subdavis) [reply docs/Pipeline-Upload.md]: @russeldj thanks I added a reference to this.  However, with this change, we will also support `includes` except in a very niche case.

**Review by BryonLewis — APPROVED** [review]

**Pull request merged**